### PR TITLE
OAuth: Support pagination for GitHub orgs

### LINKED ISSUE #58645: Github oauth provider only supports 30 organizations, prevents login for some users
**What happened**:

A new grafana installation attached to enterprise github oauth, which has login restricted to a single. One of our users was in a large number of organizations (54), and was getting login failures that he was not in the require organization. It appears that while the github teams lookup supports pagination, the github orgs function does not.

https://github.com/grafana/grafana/blob/main/pkg/login/social/github_oauth.go#L155-L179

**What you expected to happen**:

Users with > 30 organizations can log in.

**How to reproduce it (as minimally and precisely as possible)**:

Create a very large number or github orgs, put users in that, and filter based on github org. I can cause failures.

### PR DESCRIPTION
**Pull request by sdague**
This implements the same pagination strategy for github orgs as exists in the github teams.

Fixes #58645

### CODE REVIEW

File: pkg/login/social/github_oauth.go [hunk 1]
```diff
@@ -153,27 +153,33 @@ func (s *SocialGithub) HasMoreRecords(headers http.Header) (string, bool) {
 }
 
 func (s *SocialGithub) FetchOrganizations(client *http.Client, organizationsUrl string) ([]string, error) {
+	url := organizationsUrl
```
Comment (mgyongyosi) [inline pkg/login/social/github_oauth.go:156]: Could you please add the `?per_page=100` query parameter to make this coherent with the request url in the`FetchTeamMemberships` function?

**Review by mgyongyosi — APPROVED** [review]
Thank your for this PR @sdague! 🚀 I left a suggestion to make this coherent with the other function, but otherwise it looks good to me!

File: pkg/login/social/github_oauth.go (hunk 1 above)
Comment (sdague) [reply pkg/login/social/github_oauth.go:156]: Adjusted

**Comment by mgyongyosi** [comment]
Hey @sdague, thank you for the PR! It has been merged! 🎉

**Pull request merged**